eclipse-collections/src/main/java/org/eclipse/collections/impl/collector/Collectors2.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2016 Goldman Sachs.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.collector;
+
+import java.util.StringJoiner;
+import java.util.Vector;
+import java.util.stream.Collector;
+
+import org.eclipse.collections.api.bimap.MutableBiMap;
+import org.eclipse.collections.api.block.function.Function;
+import org.eclipse.collections.api.stack.MutableStack;
+import org.eclipse.collections.api.stack.StackIterable;
+import org.eclipse.collections.impl.factory.BiMaps;
+import org.eclipse.collections.impl.factory.Stacks;
+
+public final class Collectors2
+{
+    private static final Collector.Characteristics[] EMPTY_CHARACTERISTICS = {};
+    private static final Collector<?, ?, String> DEFAULT_MAKE_STRING = Collectors2.makeString(", ");
+
+    private Collectors2()
+    {
+    }
+
+    public static <T> Collector<T, ?, String> makeString()
+    {
+        return (Collector<T, ?, String>)DEFAULT_MAKE_STRING;
+    }
+
+    public static <T> Collector<T, ?, String> makeString(CharSequence separator)
+    {
+        return Collectors2.makeString("", separator, "");
+    }
+
+    public static <T> Collector<T, ?, String> makeString(CharSequence start, CharSequence separator, CharSequence end)
+    {
+        return Collector.of(
+                () -> new StringJoiner(separator, start, end),
+                (joiner, each) -> joiner.add(String.valueOf(each)),
+                StringJoiner::merge,
+                StringJoiner::toString,
+                EMPTY_CHARACTERISTICS);
+    }
+
+    public static <T> Collector<T, ?, MutableStack<T>> toStack()
+    {
+        return Collector.of(
+                Stacks.mutable::empty,
+                MutableStack::push,
+                (r1, r2) -> {r2.each(r1::push); return r1;},
```

Comment:
No, great catch.  I need to add some tests using parallelStream().collect() for each Collector.